### ISSUE TITLE
Add --keyvalue KEY=VALUE

### ISSUE THREAD

**Issue by david-a-wheeler** [boyski/pmaudit#4]
It'd be very convenient to be able to pass into pmaudit (and pmash) key/value pairs to be placed in the JSON file. E.g.:

~~~~
pmaudit --keyvalue DEPENDENCIES="foo bar" --kevalue TARGETS="makeme" ....
~~~~

These would be stored in the JSON for later use:

~~~~
  "KEYVALUE" : {
     "DEPENDENCIES: : "foo bar",
     "TARGETS" : "makeme"
  }
~~~~

That would enable me to pass data (e.g., from an execution of make) for later analysis because it would all be in the JSON file waiting for me.

**Comment by david-a-wheeler** [comment]
I'm happy to implement this, but I'd like to hear if you're okay with it first!!

**Comment by boyski** [comment]
I can't think of any objection to it. Not sure I like the name, since the
existence of a KEYDATA dict implies that the rest of the data is not
key-value which of course it is. Have you considered --custom-data and
CUSTOMDATA or something along those lines? I have no real strong feeling
though.

**Comment by david-a-wheeler** [comment]
Fair point.  However, everything is data.  I suggest "--custom" for the option to supply custom data.

**Comment by boyski** [comment]
As noted in previous email argparse does auto-abbreviation so the user
could use --custom in either case, but I agree that "data" adds little
value so --custom is fine with me.

**Comment by david-a-wheeler** [comment]
I've implemented --custom in the pmaudit pull request #1 .  Eventually it should show up in pmash.c, but pmash will first have to learn to generate JSON at all :-).  I want to focus on getting pmaudit having the right functionality & interface... we can then re-implement what's needed in pmash.c.
